bin/test/main.ml

```diff
@@ -234,95 +250,65 @@ let run_exn (`Setup ()) (`Non_deterministic non_deterministic)
-            let eval () =
-              eval_raw Block.empty ?root c ~line:0 (Mdx.Util.File.read_lines f)
-            in
-            match env with None -> eval () | Some e -> Mdx_top.in_env e eval)
-          fs
-    | _ -> Fmt.failwith "only one of --prelude or --prelude-str shoud be used"
-  in
+  eval_prelude ?root c prelude prelude_str;
 
   let test_block ~ppf ~temp_file t =
-    let active = active t && Block.version_enabled t && not (Block.skip t) in
     let print_block () = Block.pp ?syntax ppf t in
-    match
-      (active, non_deterministic, Block.file t, Block.mode t, Block.value t)
-    with
-    (* Print errors *)
-    | _, _, _, _, Error _
-    (* The command is not active, skip it. *)
-    | false, _, _, _, _
-    (* the command is active but non-deterministic so skip everything *)
-    | true, false, _, `Non_det Nd_command, _ ->
-        print_block ()
-    (* the command is active but it's output is
-       non-deterministic; run it but keep the old output. *)
-    | true, false, _, `Non_det Nd_output, Cram { tests; _ } ->
-        print_block ();
-        let blacklist = Block.unset_variables t in
-        List.iter
-          (fun t ->
-            let (_ : int) = run_test ?root blacklist temp_file t in
-            ())
-          tests
-    | true, false, _, `Non_det Nd_output, Toplevel tests ->
-        assert (syntax <> Some Cram);
-        print_block ();
-        List.iter
-          (fun test ->
-            match eval_test t ?root c test with
-            | Ok _ -> ()
-            | Error e ->
-                let output = List.map (fun l -> `Output l) e in
-                if Output.equal test.output output then ()
-                else err_eval ~cmd:test.command e)
-          tests
-    | true, _, Some ext_file, _, kind -> (
-        match kind with
-        | OCaml | Toplevel _ ->
-            assert (syntax <> Some Cram);
-            update_file_or_block ?root ppf file ext_file t
-        | _ when Util.Option.is_some (Block.part t) ->
-            Fmt.failwith "Parts are not supported for non-OCaml code blocks."
-        | Cram _ | Raw ->
-            let new_content = read_part ext_file (Block.part t) in
-            update_block_content ppf t new_content
-        | _ -> print_block () )
-    | true, _, None, _, kind -> (
-        match kind with
-        | OCaml ->
-            assert (syntax <> Some Cram);
-            eval_raw t ?root c ~line:t.line t.contents;
-            Block.pp ppf t
-        | Cram { tests; pad } ->
-            run_cram_tests ?syntax t ?root ppf temp_file pad tests
-        | Toplevel tests ->
-            assert (syntax <> Some Cram);
-            run_toplevel_tests ?root c ppf tests t
-        | Raw | _ -> print_block () )
+    if Block.is_active ?section t then
+      match Block.value t with
+      (* Print errors *)
+      | Error _ -> print_block ()
+      | Raw _ -> print_block ()
+      | Include { file_included; part_included; header } -> (
```

Comment:
Do you mean only for Include blocks or for all block types?

Reply:
Just for file includes here, since the header is still part of the parent record type, we have all the information needed to print it so it's okay to remove it from the `Include` record arg.

As a matter of fact, since parts are only meaning full for OCaml files, how about having a record type that don't allow construction of forbiden values by merging `part` and `header` into a single field that could be `Ocaml of part option | Other`?

Reply:
That's something that could be explored in a subsequent PR, I think the diff is already big enough

Reply:
To be honest I think this falls exactly into the scope of this PR, ie improving the block representation so that errors are dealt with early and we don't have to care about inconsistent values after the parsing is done.

Reply:
done